funlet-simple-message/functions/funlet-simple-message.js

```diff
@@ -0,0 +1,210 @@
+      env.FUNLET_MESSAGE3,
+      env.FUNLET_MESSAGE4,
+      env.FUNLET_MESSAGE5
+    ].filter( message => typeof message === "string" );
+  }
+  return config.messages;
+}
+
+function getLanguage(params, env, config) {
+  return params.Language || env.FUNLET_MESSAGE_LANGUAGE || config.language;
+}
+
+function getVoice(params, env, config) {
+  return params.Voice || env.FUNLET_MESSAGE_VOICE || config.voice;
+}
+
+/*
+  3. Output Helpers
+
+  These helper functions build part of the output.
+
+  This is where you can fine-tune the TwiML elements and attributes
+  produced in response to each stage of the Funlet.
+*/
+
+/*
+  Function: simpleMessage()
+
+  Parameters:
+    * response - Twilio.twiml.VoiceResponse, Twilio Voice response in progress
+    * message - string, recorded message (URL starting with 'http') to play
+                or text message to say
+    * language - string, language for text messages, e.g. 'en' for English
+                 with an American accent or 'en-gb' for English with a British
+                 accent. Use the voice 'alice' for the largest list of
+                 supported languages and accents.
+    * voice - string, voice for text messages,
+              one of 'man', 'woman' or 'alice'.
+
+  Response:
+    The input response is modified with further instructions to Play or Say
+    the given message. The input response is returned unchanged when the
+    given message is empty.
+*/
+function simpleMessage(response, message, language, voice) {
+  if ( message.length === 0 ) {
+    return;
+  }
+  if ( message.startsWith("http") ) {
+    response.play({}, message);
+  } else {
+    response.say({language:language, voice:voice}, message);
+  }
+}
+
+/*
+  Function: simpleMessages()
+
+  Parameters:
+    * response - Twilio.twiml.VoiceResponse, Twilio Voice response in progress
+    * messages - array of string, list of messages, either recorded messages
+                 (URL starting with 'http') to play or text message to say
+    * language - string, language for text messages, e.g. 'en' for English
+                 with an American accent or 'en-gb' for English with a British
+                 accent. Use the voice 'alice' for the largest list of
+                 supported languages and accents.
+    * voice - string, voice for text messages,
+              one of 'man', 'woman' or 'alice'.
+
+  Response:
+    The input response is modified with further instructions to Play or Say
+    each message. The input response is returned unchanged when the
+    list of messages is empty.
+*/
+function simpleMessages(response, messages, language, voice) {
+  messages.forEach(
+    message => simpleMessage(response, message, language, voice)
+  );
+}
+
+/*
+  4. Main Handler
+
+  This is the entry point to your Twilio Function,
+  which will run to process an incoming HTTP request
+  such as the ones generated by Twilio events.
+*/
+
+exports.handler = function(env, params, reply) {
```

Comment:
I realise this is the pattern across all the funlets, but I would prefer to see the handler use the same signature as all the other functions in this repo, which is:

```js
function(context, event, callback) {
```

This also matches the documentation on twilio.com.

Reply:
I understand. This is my pet peeve, I admit :) I just find the official way confusing (I keep stumbling on `event` vs `env`, and how context is `env` but comes before the params), but I will make the change for consistency, no problem.